app/src/ui/history/commit-list-item.tsx

```diff
@@ -79,12 +81,33 @@ export class CommitListItem extends React.Component<
             </div>
           </div>
         </div>
+        {this.renderUnpushedIndicator()}
       </div>
     )
   }
 
   public shouldComponentUpdate(nextProps: ICommitProps): boolean {
-    return this.props.commit.sha !== nextProps.commit.sha
+    return (
+      this.props.commit.sha !== nextProps.commit.sha ||
+      this.props.showUnpushedIndicator !== nextProps.showUnpushedIndicator
+    )
```

Comment:
i appreciate the restraint here :) or at least, i would have felt tempted to refactor this `shouldComponentUpdate` away, but i think its better not to in this PR